domains/emissions/Emissions.API/Tests/CalculateSourceEmissionShareTest.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using API.Helpers;
+using API.Models;
+using API.Services;
+using Xunit;
+
+namespace Tests
+{
+    public sealed class CalculateSourceEmissionShareTest
+    {
+        readonly SourceEmissionShareDataSetFactory sourceEmissionShareDataSetFactory = new();
+
+        [Theory]
+        [InlineData(Aggregation.Total)]
+        [InlineData(Aggregation.Actual)]
+        [InlineData(Aggregation.Hour)]
+        [InlineData(Aggregation.Day)]
+        [InlineData(Aggregation.Month)]
+        [InlineData(Aggregation.Year)]
+        public void EmissionSharesAndMeasurements_CalculateTotalEmission_TotalAnRelativeEmission(Aggregation aggregation)
+        {
+            // Arrange
+            var dateFrom = new DateTime(2021, 1, 1, 22, 0, 0, DateTimeKind.Utc);
+            var dateTo = new DateTime(2021, 1, 2, 1, 59, 59, DateTimeKind.Utc);
+            var timeSeries = sourceEmissionShareDataSetFactory.CreateTimeSeries();
+            var emissionShares = sourceEmissionShareDataSetFactory.CreateEmissionsShares();
+            Environment.SetEnvironmentVariable("RENEWABLESOURCES",
+                "Wood,Waste,Straw,BioGas,Solar,WindOnshore,WindOffshore");
+
+
+            var sut = new SourcesCalculator();
+
+            // Act
+            var result = sut.CalculateSourceEmissions(timeSeries, emissionShares, aggregation);
+
+            //Assert
+            Assert.NotNull(result);
+            var emissionsEnumerable = GetExpectedSourceEmissions(aggregation, dateFrom, dateTo).EnergySources;
+            Assert.Equal(emissionsEnumerable.Select(_ => _.Renewable), result.EnergySources.Select(_ => _.Renewable));
+            Assert.Equal(emissionsEnumerable.Select(_ => _.Ratios), result.EnergySources.Select(_ => _.Ratios));
+            Assert.Equal(emissionsEnumerable.Select(_ => _.DateFrom), result.EnergySources.Select(_ => _.DateFrom));
+            Assert.Equal(emissionsEnumerable.Select(_ => _.DateTo), result.EnergySources.Select(_ => _.DateTo));
+        }
+
+        EnergySourceResponse GetExpectedSourceEmissions(Aggregation aggregation, DateTime dateFrom, DateTime dateTo)
```

Comment:
```suggestion
        private EnergySourceResponse GetExpectedSourceEmissions(Aggregation aggregation, DateTime dateFrom, DateTime dateTo)
```

Reply:
This is private, when nohting have been specified.